app/controllers/thread_controller.php

```diff
@@ -0,0 +1,62 @@
+<?php
+class ThreadController extends AppController 
+{
+    function __construct($name)
+    {
+        parent::__construct($name);
+
+       if (!is_logged_in())
```

Comment:
Edited. Thank you.

Update
As mentioned before, the constructor for this class was removed. Thank you.